plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcIdentityCacheMapping.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+public interface JdbcIdentityCacheMapping
+{
+    JdbcIdentityCacheKey getRemoteUserCacheKey(JdbcIdentity identity);
+
+    abstract class JdbcIdentityCacheKey
```

Comment:
I am not happy with the resolution to https://github.com/trinodb/trino/pull/6803#discussion_r571969380 .

As i said, adding hashCode and equals would be necessary -- and you did that, thanks -- but I also believe this is an over-engineered approach, and also overly fragile, where as a simple `boolean` flag would be sufficient.

Reply:
> I also believe this is an over-engineered approach, and also overly fragile, where as a simple boolean flag would be sufficient.

@findepi I hear your voice, also I see issues you are mentioning. Fair points. However I don't see a simple solution here. Can you please help me to model this?

There are following cases:
 - Trino session identity does matter, as we are using same credentials to access remote data source. Let it it be `false`.
 - Trino session identity matters, as we are using extra credentials to access data from remote data source as different user. Let it it be `true`. Notice that here not entire identity matters, but only these extra credentials that are used to connect to remote data source.
 - Trino session identity matters, as we might use something else to access data from remote data source as different user. For example in tests I used user name. How should I map this case to `boolean` value?

Reply:
Perhaps this could be something like:

- Trino session identity matters <=> use JdbcIdentity as the cache key

Reply:
It fails with:

> The fact that user sends in GCP's GS related extra credentials should not affect caching.

Where unrelated things in `JdbcIdentity` could make caching ineffective.

Reply:
I i read the force push diff (https://github.com/trinodb/trino/compare/b11c6483f13fd2548603eb3e98b170c3368cc032..b48c4a910c02a1534cc832b42fb786bdda0f8177) correctly, when I wrote that comment two weeks ago the implementation would pull in all the extra credentials, so i was probably referring to this.

Anyway, i am considering that we don't need this complexity, and we can just have "Trino session identity matters <=> use JdbcIdentity as the cache key".

If we consider we _do_ need to discern extra credentials, let's session properties handling be the guiding thought.

For example in this PR we have

```
ColumnsCacheKey key = new ColumnsCacheKey(getIdentityKey(session), getSessionProperties(session), tableHandle.getRequiredNamedRelation().getSchemaTableName());
```
 and then 
```
private Map<String, Object> getSessionProperties(ConnectorSession session)
{
    return sessionProperties.stream()
            .map(property -> Map.entry(property.getName(), getSessionProperty(session, property)))
            .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
}
```

_If_ we need to discern extra credentials, we could do likewise -- provide the list of credential names that matter to the `CachingJdbcClient`.

Reply:
> when I wrote that comment two weeks ago the implementation would pull in all the extra credentials, so i was probably referring to this.

Yes, it you are correct. Things were different, but then I addressed: https://github.com/trinodb/trino/pull/6803#discussion_r571967588

It is not only about extra credentials. Depending on connectors capabilities, it could be different. Sometimes it could be even just user name.